react-components/src/components/NodeCacheProvider/AssetMappingCache.ts

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { type CogniteClient, type AssetMapping3D, type Node3D } from '@cognite/sdk';
+import { type ModelId, type ModelRevisionKey, type RevisionId } from './types';
+import { maxBy } from 'lodash';
+import assert from 'assert';
+
+export type NodeAssetMappingResult = { node?: Node3D; mappings: AssetMapping3D[] };
+
+export class AssetMappingCache {
+  private readonly _sdk: CogniteClient;
+
+  private readonly _modelToAssetMappings = new Map<ModelRevisionKey, AssetMapping3D[]>();
```

Comment:
Would it be more efficient to use a map instead of an array for asset mappings?

Reply:
Hmm, in case we need to lookup asset mappings, you mean? That's true :thinking: But I think I'll postpone that decision until it's needed